instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/event_handler.rb

```diff
@@ -191,11 +193,12 @@ def request_span_attributes(env)
             attributes
           end
 
-          def detach_contexts(request)
-            request.env[TOKENS_KEY]&.reverse_each do |token|
-              OpenTelemetry::Context.detach(token)
-              OpenTelemetry::Trace.current_span.finish
-            end
+          def detach_context(request)
+            return nil unless request.env[RACK_OTEL_CONTEXT_KEY]
+
+            token, rack_ctx = request.env[RACK_OTEL_CONTEXT_KEY]
+            OpenTelemetry::Trace.current_span(rack_ctx).finish
+            OpenTelemetry::Context.detach(token)
```

Comment:
If we store the token and span instead:
```suggestion
            return nil unless request.env[OTEL_TOKEN_AND_SPAN]

            token, span = request.env[OTEL_TOKEN_AND_SPAN]
            span.finish
            OpenTelemetry::Context.detach(token)
```

Reply:
👍 https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/965/commits/dfcbfb257e2c59965e2f0ee191c523fe019cf0bc